### PR TITLE
add apilisting service listing watch

### PR DESCRIPTION
**Pull request by tharindu1st**
add apilisting service listing watch

### CODE REVIEW

File: runtime/runtime-domain-service/ballerina/Init.bal [hunk 1]
```diff
@@ -17,11 +17,18 @@
...
+import ballerina/task;
 
 listener http:Listener ep0 = new (9443);
 
 # Initializing method for runtime
 function init() {
     log:printInfo("Initializing Runtime Domain Service..");
+    do {
+        _ = check task:scheduleJobRecurByFrequency(new ServiceTask(), 1);
+        // _ = check task:scheduleJobRecurByFrequency(new APIListingTask(), 1);
```
Comment (Krishanx92) [inline runtime/runtime-domain-service/ballerina/Init.bal:29]: is this commented by mistake?

File: runtime/runtime-domain-service/ballerina/K8sClient.bal [hunk 2]
```diff
@@ -45,132 +46,76 @@ function initializeK8sClient() returns http:Client|error {
...
-            APINames.push(APIData);
-        }
-        APIList APIList = {
-            list: APINames
-        };
-        return APIList;
+function getAPIList() returns APIList|error {
+    API[] apilist = [];
+    foreach model:K8sAPI api in getAPIs() {
+        API convetedModel = convertK8sAPItoAPI(api);
```
Comment (AmaliMatharaarachchi) [inline runtime/runtime-domain-service/ballerina/K8sClient.bal:86]:
```suggestion
        API convertedModel = convertK8sAPItoAPI(api);
```

File: runtime/runtime-domain-service/ballerina/main.bal [hunk 3]
```diff
@@ -0,0 +1,101 @@
+import ballerina/http;
+import ballerina/log;
+
+final http:Client clientEP =
+check new ("https://192.168.49.2:8443",
+auth = {
+    token: "eyJhbGciOiJSUzI1NiIsImtpZCI6IkszUThKSWZWVUR0dWc4TnE1bEYwZS1jbmh5UTh4UEg1TlFVUjlTbmdTUjQifQ.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJhcGstcGxhdGZvcm0iLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlY3JldC5uYW1lIjoid3NvMmFway1wbGF0Zm9ybS10b2tlbi10OGJkdCIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50Lm5hbWUiOiJ3c28yYXBrLXBsYXRmb3JtIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQudWlkIjoiOGNlNmQ1ZjMtZjY5Yi00NDNkLWE1NzAtZThiOGEwZjdhNTQ1Iiwic3ViIjoic3lzdGVtOnNlcnZpY2VhY2NvdW50OmFway1wbGF0Zm9ybTp3c28yYXBrLXBsYXRmb3JtIn0.e8wCP9RhHbuykbFZoyQ-SC7k7XCLXxVAiHFB_LfKsc4p9cQseVuO2Z2r-RIeuNKzdjM79VhgehJyAxZ9IxSXQZPjS6nqW_jThi94L7_JAZ7HNvsWZxJuGKKFhz4cEmWyNK8dwiGfEkk6AvyW0tm2QEJjaP6SEEinodLjYE17sdOla3ABO-HIDNY-SpYH4hWx_5-gmNihDZm3ME82DgFFX3bf7ayJOheV7scEnaZyC2KKqFVLpHQhO_H7psDCJ2kr198w56Zvsmx10IBNIKHP-YwrnNqwbu1OEV94lg7qYav87zO9s3ON49WRLGHad-oNqZmTKd2OpT2qnOM_GkQHQw"
+},
+    secureSocket = {
+        cert: "/home/tharindud/Downloads/minikubeCA.cer"
```
Comment (AmaliMatharaarachchi) [inline runtime/runtime-domain-service/ballerina/main.bal:10]: better to make them configurable, isn't it

Comment (tharindu1st) [reply runtime/runtime-domain-service/ballerina/main.bal]: this is removed already

File: runtime/runtime-domain-service/ballerina/Init.bal (hunk 1 above)
Comment (tharindu1st) [reply runtime/runtime-domain-service/ballerina/Init.bal]: done

**Review by Krishanx92 — APPROVED** [review]

**Review by Amila-Rukshan — APPROVED** [review]

**Pull request merged**